### PR TITLE
Added Xamarin Metrics Documentation

### PR DESCRIPTION
**Pull request by ziccardi**
Added documentation about metrics for the Xamarin platform.

Issue: https://issues.jboss.org/browse/AEROGEAR-2864

### CODE REVIEW

File: modules/ROOT/pages/xamarin/using-metrics-sdk.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+= AeroGear Services Metrics SDK
+
+This document explains how to use the AeroGear Metrics SDK on the Android platform and integrate it with an App.
```
Comment (jstaffor) [inline modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: This file is located within the context of 'xamarin' yet 'Android' is mentioned in the title. Should the state: 'xamarin' ?

File: modules/ROOT/pages/xamarin/using-metrics-sdk.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+== Provided metrics
+
+The default metrics contain App metrics, Device metrics and possibly other types in the future.
```
Comment (jstaffor) [inline modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: best to remove this "and possibly other types in the future" so it reads something like:
 - "The default metrics contain mobile app and mobile device metrics"

File: modules/ROOT/pages/xamarin/using-metrics-sdk.adoc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+A Client ID and a timestamp are added to the top level of the Metrics object for categorization and visualization purposes.
+The Client ID will be generated and stored on the device when the App is started the first time.
```
Comment (jstaffor) [inline modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: When the App is started the for first time, a Client ID is generated and stored on the mobile device

**Comment by ziccardi** [comment]
@jstaffor Mind giving another look?

File: modules/ROOT/pages/xamarin/using-metrics-sdk.adoc [hunk 4]
```diff
@@ -0,0 +1,24 @@
+= AeroGear Services Metrics SDK
```
Comment (finp) [inline modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: @johnfriz  thought you said there isn't a metrics sdk ;)

Comment (johnfriz) [reply modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: I guess I was wrong!

@wei-lee - Is this a Xamarin only thing? I thought metrics were a part of core?

Comment (wei-lee) [reply modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: @johnfriz It is part of the core. I think the title here is wrong. It should be part of the core module.

Comment (wei-lee) [reply modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: We have created similar files for all other platforms, like https://github.com/aerogear/mobile-docs/blob/master/modules/ROOT/pages/ios/using-metrics-sdk.adoc and https://github.com/aerogear/mobile-docs/blob/master/modules/ROOT/pages/android/using-metrics-sdk.adoc. I think they all need to be updated.

File: modules/ROOT/pages/xamarin/security.adoc [hunk 5]
```diff
@@ -186,3 +186,23 @@ Dictionary<string, SecurityCheckResult> results = securityService.GetSyncExecuto
 
 The `Builder.WithSecurityCheck` method can be invoked as much as needed to add checks to be executed.
 The value returned by the `Execute` method will be a `Dictionary` whose value is the result of the check, while the key is the ID of the check that produced that result.
+
+[#reporting-selfdefence-metrics]
+=== Reporting Self Defence Checks Results Via the Metrics Service
+In order to report the results of Self Defence utilize the link:./using-metrics-sdk[Metrics] service in conjunction with the `SecurityService`.
```
Comment (jstaffor) [inline modules/ROOT/pages/xamarin/security.adoc]: "In order to report the results of Self Defence **checks**, **use**

Metrics **S**ervice

service in conjunction with the `SecurityService`  - should it be 'ISecurityService'

File: modules/ROOT/pages/xamarin/security.adoc [hunk 6]
```diff
@@ -186,3 +186,23 @@ Dictionary<string, SecurityCheckResult> results = securityService.GetSyncExecuto
...
+
+Reporting the check results can be done via an `Executor` that is passed a reference to the `MetricsService`:
```
Comment (jstaffor) [inline modules/ROOT/pages/xamarin/security.adoc]: Reporting the check results **is** done **using** an

File: modules/ROOT/pages/xamarin/using-metrics-sdk.adoc [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+This document explains how to use the AeroGear Metrics SDK on the Xamarin platform and integrate it with an App.
```
Comment (jstaffor) [inline modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: This **section** explains

File: modules/ROOT/pages/xamarin/using-metrics-sdk.adoc [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+== Provided metrics
+
+The default metrics contain mobile app and mobile device metrics.
```
Comment (jstaffor) [inline modules/ROOT/pages/xamarin/using-metrics-sdk.adoc]: This isn't clear - does the he default metrics **service**/**sdk**/**something else** contain a mobile and a mobile device metrics?

**Comment by ziccardi** [comment]
@jstaffor I addressed your comments. Mind giving another look?

**Pull request merged**